hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -553,7 +553,7 @@ private void writeChunk(ChunkBuffer buffer)
   /**
    * @param close whether the flush is happening as part of closing the stream
    */
-  private void handleFlush(boolean close)
+  protected void handleFlush(boolean close)
```

Comment:
Let's rename the current `handleFlush` to `handleFlushInternal` and then add a new `handleFlush` with try-catch as below.
```java
@@ -553,7 +544,31 @@ private void writeChunk(ChunkBuffer buffer)
   /**
    * @param close whether the flush is happening as part of closing the stream
    */
-  private void handleFlush(boolean close)
+  void handleFlush(boolean close) throws IOException{
+    try {
+      handleFlushInternal(close);
+    } catch (ExecutionException e) {
+      // just set the exception here as well in order to maintain sanctity of
+      // ioException field
+      handleExecutionException(e);
+    } catch (InterruptedException ex) {
+      Thread.currentThread().interrupt();
+      handleInterruptedException(ex, true);
+    } catch (Throwable e) {
+      String msg = "Failed to flush. error: " + e.getMessage();
+      LOG.error(msg, e);
+      throw e;
+    } finally {
+      if (close) {
+        cleanup(false);
+      }
+    }
+  }
+
+  private void handleFlushInternal(boolean close)
       throws IOException, InterruptedException, ExecutionException {
     checkOpen();
     // flush the last chunk data residing on the currentBuffer
```